lib/tlssock.h

```diff
@@ -29,14 +29,40 @@
 #define TLS_CLT_HANDSHAKE 1
 #define TLS_SRV_HANDSHAKE 2
 
+#define TLS_OPT_PEER_SUBJECT_DN 1
+
+#define TLS_CLIENT_CERT_IGNORE 0
+#define TLS_CLIENT_CERT_REQUEST 1
+#define TLS_CLIENT_CERT_REQUIRE 2
+
 typedef struct {
   void *misc;
 
   ssize_t (*psk)(void *misc, char **username, uint8_t **key);
+  struct {
+    int (*getcert)(void *misc,
+                const char **requested_ca_dn,
+                char **cert_uri,
+                char **key_uri,
+                char **pin);
+    const char *cafile;
+    int insecure;
+    const char *hostname;
+  } cert;
 } tls_clt_handshake_t;
 
 typedef struct {
   void *misc;
 
   ssize_t (*psk)(void *misc, const char *username, uint8_t **key);
+  struct {
+    int (*getcert)(void *misc,
+                const char *servername,
+                char **cert_uri,
+                char **key_uri,
+                char **pin);
+    int client_certificate_request;  // 0 for no client cert, 1 for optional, 2 for required
+    const char *cafile;
```

Comment:
If I understand this correctly, we can use the implicit value of `cafile` to determine if a client certificate can be validated. This means that `client_certificate_request` can become `bool client_certificate_required`.